### PR TITLE
Fix ScrollView onScroll event only being called once.

### PR DESCRIPTION
**Pull request by chrene**

### CODE REVIEW

**Comment by tadeuzagallo** [comment]
@nicklockwood I think you wrote the TODO, is that right?
I don't think 0 should be the default value...

**Comment by tadeuzagallo** [comment]
@nicklockwood I think you wrote the TODO, is that right?
I don't think 0 should be the default value...

**Comment by sahrens** [comment]
We don't want to send the events by default, and we want the client to think about what frequency they need the events. If the parent specifies onScroll and not the frequency, perhaps we should redbox?

**Comment by tadeuzagallo** [comment]
@sahrens it might have been my fault, I was talking @chrene on IRC and he said it was a bug, and I totally forgot that we needed to set the property for it to work.
If we set the property is working as expected? Because if it is maybe we should remove that comment.

**Comment by chrene** [comment]
The only real bug was just that it was a double when it should have been a NSUInteger. Since it was a double it got a very very high number when set. Like 4 billion or something. Now it gets the real value. We can keep the check in the if statement such that the client needs to specify another value than 0.

File: React/Views/RCTScrollView.m [hunk 1]
```diff
@@ -404,9 +405,9 @@ - (void)scrollViewDidScroll:(UIScrollView *)scrollView
   if (_allowNextScrollNoMatterWhat ||
       (_throttleScrollCallbackMS != 0 && throttleScrollCallbackSeconds < (now - _lastScrollDispatchTime))) {
 
-    // Calculate changed frames
-    NSMutableArray *updatedChildFrames = [[NSMutableArray alloc] init];
-    [[_contentView reactSubviews] enumerateObjectsUsingBlock:^(UIView *subview, NSUInteger idx, BOOL *stop) {
+      // Calculate changed frames
+      NSMutableArray *updatedChildFrames = [[NSMutableArray alloc] init];
+      [[_contentView reactSubviews] enumerateObjectsUsingBlock:^(UIView *subview, NSUInteger idx, BOOL *stop) {
```
Comment (tadeuzagallo) [inline React/Views/RCTScrollView.m]: nit: this indentation was right...

File: React/Views/RCTScrollView.m [hunk 2]
```diff
@@ -395,6 +395,7 @@ - (void)scrollViewDidScroll:(UIScrollView *)scrollView
   NSTimeInterval now = CACurrentMediaTime();
   NSTimeInterval throttleScrollCallbackSeconds = _throttleScrollCallbackMS / 1000.0;
 
+
```
Comment (tadeuzagallo) [inline React/Views/RCTScrollView.m]: nit: remove this line

**Comment by tadeuzagallo** [comment]
I tested it and it's working as expected, but @sahrens I tried adding an invariant instead of the current warning, and during testing it on UIExplorer I just noticed that an invariant during a NavigatorIOS push crashes the objC... Any idea about that? Should we add the invariant anyways?

**Comment by sahrens** [comment]
If there is already a warning we can leave it.

**Comment by nicklockwood** [comment]
I submitted a fix for this today that changes the property type to NSTimeInterval and renames the property to scrollEventThrottle, which seems more logical. That also fixes the issue with the default value being set incorrectly.

**Comment by sahrens** [comment]
This should be fixed in master - please open an issue if still see the bug.

**Pull request closed without merging**